dpe/src/commands/derive_context.rs

```diff
@@ -262,11 +284,16 @@ impl CommandExecution for DeriveContextCmd {
                         // Should be ignored since retain_parent cannot be true
                         parent_handle: ContextHandle::default(),
                         resp_hdr: ResponseHdr::new(DpeErrorCode::NoError),
+                        exported_cdi: [0; MAX_EXPORTED_CDI_SIZE],
+                        certificate_size: 0,
+                        new_certificate: [0; MAX_CERT_SIZE]
```

Comment:
Currently in the spec, these fields are only returned if `EXPORT_CDI == true`: https://github.com/TrustedComputingGroup/Server-Internal/blob/main/dpe-irot-profile/main.tcg#L1450-L1471

So I think we would want to have two different response structs and conditionally device which one to return. Although given how the Response enum works, that might be a bit of a pain. So maybe it's better to just cut that part out at a higher level.

wdyt?

Reply:
I need to think deeper on this but maybe we could achieve something similar wrapping these fields with `Option<T>`.

That at least solves the serialization (I think).

Reply:
In hindsight, I don't think using `Option<T>` will work.

> So maybe it's better to just cut that part out at a higher level.

I agree, a single response will be easier.

Reply:
Closing since this is now handled with a speparate struct.